Add quality declaration console_bridge_vendor (#8)
ros2/console_bridge_vendor: Quality_Declaration.md

> This document is a declaration of software quality for the `console_bridge_vendor` package, based on the guidelines in [REP-2004](https://www.ros.org/reps/rep-2004.html).

Reviewer: I just noticed how you were naming all of these, I thought we were naming things QUALITY_DECLARATION.md?

Author: Yes, we agreed on the standard after this was opened initially, I'll fix it for all my PRs.

Author: Just checked, the latest commit of this PR already has this addressed.

# `console_bridge_vendor` Quality Declaration

The package `console_bridge_vendor` claims to be in the **Quality Level 4** category.

Below are the rationales, notes, and caveats for this claim, organized by each requirement listed in the [Package Quality Categories in REP-2004](https://index.ros.org/doc/ros2/Contributing/Developer-Guide/#package-quality-categories) of the ROS2 developer guide.

## Version Policy [1]

### Version Scheme [1.i]

`console_bridge_vendor` uses `semver` according to the recommendation for ROS Core packages in the [ROS 2 Developer Guide](https://index.ros.org/doc/ros2/Contributing/Developer-Guide/#versioning).

### Version Stability [1.ii]

`console_bridge_vendor` is at or above a stable version, i.e. `>= 1.0.0`.

### Public API Declaration [1.iii]

> External dependency is loaded downloaded from upstream and there is no public API declaration hosted for `libconsole-bridge-dev`.

Reviewer: ```suggestion
This is a vendor package for `libconsole-bridge-dev` and as such does not declare its own API.
```

### API Stability Policy [1.iv]

> `console_bridge_vendor` is pinned to a `libconsole-bridge-dev` version (Currently 0.4.1), so it will not break public API within a released ROS distribution, i.e. no major releases once the ROS distribution is released.

Reviewer: This sentence is not clear. Try something like:

"`console_bridge_vendor` does not have an API of its own, and will limit its dependency on `libconsole-bridge-dev` to API stable releases within a released ROS distribution."

### ABI Stability Policy [1.v]

> `console_bridge_vendor` does not have an ABI of its own, and will limit its dependency on `libconsole-bridge-dev` to API stable releases within a released ROS distribution.

Reviewer: ```suggestion
`console_bridge_vendor` does not have an ABI of its own, and will limit its dependency on `libconsole-bridge-dev` to ABI stable releases within a released ROS distribution.
```

> `console_bridge_vendor` is pinned to a `libconsole-bridge-dev` version (Currently 0.4.1), so it will not break public ABI within a released ROS distribution, i.e. no major releases once the ROS distribution is released.

Reviewer: Same.

### ABI and ABI Stability Within a Released ROS Distribution [1.vi]

> `console_bridge_vendor` will not break API nor ABI within a released ROS distribution, i.e. no major releases once the ROS distribution is released.

Reviewer 2: I would rephrase this, if there is no API or ABI how are we going to break it?. I would say something like: "As there are not API or ABI issue in this package, this section doesn't applied"

### Change Requests [2.i]

All changes will occur through a pull request, check [ROS 2 Developer Guide](https://index.ros.org/doc/ros2/Contributing/Developer-Guide/#change-control-process) for additional information.

### Contributor Origin [2.ii]

This package uses DCO as its confirmation of contributor origin policy. More information can be found in [CONTRIBUTING](../CONTRIBUTING.md).

### Peer Review Policy [2.iii]

All pull requests will be peer-reviewed, check [ROS 2 Developer Guide](https://index.ros.org/doc/ros2/Contributing/Developer-Guide/#change-control-process) for additional information.

### Continuous Integration [2.iv]

This package is not being tested under ROS2 CI.

###  Documentation Policy [2.v]

All pull requests must resolve related documentation changes before merging.

## Documentation [3]

### Feature Documentation [3.i]

There is some documentation on its latest [wiki](http://wiki.ros.org/console_bridge). This information might require to be updated and expanded further for the use of `console_bridge_vendor`.

### Public API Documentation [3.ii]

> `console_bridge_vendor` does not have Public API Documentation.

Reviewer: "`console_bridge_vendor` does not have an API and therefore does not require API documentation."

### License [3.iii]

The license for `console_bridge_vendor` is Apache 2.0, and a summary is in each source file, the type is declared in the [`package.xml`](./package.xml) manifest file, and a full copy of the license is in the `LICENSE` file.

### Copyright Statements [3.iv]

The copyright holders each provide a statement of copyright in each source code file in `console_bridge_vendor`.

> ## Testing [4]

Reviewer: Since this is such a basic package, I would say that that feature testing, api testing, code coverage and performance testing are not needed. Otherwise, it kind of looks like they are missing, when they are really just irrelevant.

Author: I agree to that for vendored packages, as the dependencies should address this (or argue why is not needed).

### Feature Testing [4.i]

> `console_bridge_vendor` does not have features testing. For vendored packages, this will be considered to be addressed in the external dependency.

Reviewer: ```suggestion
`console_bridge_vendor` does not have feature testing. For vendored packages, this will be considered to be addressed in the external dependency.
```

### Public API Testing [4.ii]

`console_bridge_vendor` does not have Public API testing.

### Coverage [4.iii]

`console_bridge_vendor` does not provide coverage testing.

### Performance [4.iv]

`console_bridge_vendor` does not provide performance testing.

### Linters and Static Analysis [4.v]

`console_bridge_vendor` is not being tested with linters or static analysis tools.

## Dependencies [5]

### Direct and Optional Runtime ROS Dependencies [5.i]/[5.ii]

`console_bridge_vendor` does not have direct/optional runtime ROS dependencies.

### Direct Runtime non-ROS Dependency [5.iii]

> `console_bridge_vendor` depends directly on the external dependency `libconsole-bridge-dev`, this one is qualified as quality level 4 as described thoroughly [QD libconsole-bridge-dev](https://github.com/ros/console_bridge/blob/master/Quality_Declaration.md).

Reviewer: ```suggestion
`console_bridge_vendor` depends directly on the external dependency `libconsole-bridge-dev`, which is qualified as quality level 4 in its [Quality Declaration](https://github.com/ros/console_bridge/blob/master/Quality_Declaration.md).
```
